### PR TITLE
Option for showing bookmarks or recent files in appicons menu

### PR DESCRIPTION
**Pull request by maoschanz**
related to issue #671 

i didn't find a .pot file so i didn't try to add new strings to translation-related files

### CODE REVIEW

**Comment by micheleg** [comment]
Hi,

thanks for the contribution and sorry for the late reply. I quickly tested this feature and seems to be working just fine. I haven't thought about it enough, but it seems to be adding a reasonable funcitonality in a reasonably uninstrusive way.

I'm jsut a bit unconvinced about the bookmarks:
image

it's not obvous to me what those icons are and I'm not convinced bookmarks should be split  between icons and list. 

I've also noticed a little bug: the menu is not closed when you click on one of those icon, while it is when clicking on the bookmark in the list (like any other menu item).

Kinf of provocative question: is the bookmark menu any faster/more direct/more logical than just opening nautilus and using the sidebar?

**Comment by Maestroschan** [comment]
>it's not obvous to me what those icons are and I'm not convinced bookmarks should be split between icons and list.

These 3 are "recent files", "trash" and "other locations", and since:
- they are not actually bookmarks;
- they are probably among the most useful items

... I didn't want to put them in the same submenu. But i agree the "other locations" icon is far from perfect

>I've also noticed a little bug: the menu is not closed when you click on one of those icon, while it is when clicking on the bookmark in the list (like any other menu item).

Exact, i'll fix that

>is the bookmark menu any faster/more direct/more logical than just opening nautilus and using the sidebar?

Faster: if Nautilus is already launched, clicking on the sidebar is faster, but otherwise it's not

More direct/logical: yes, imo at least, and it's what Unity and Windows Vista to 10 are doing (no idea what macOS does)

File: appIcons.js [hunk 1]
```diff
@@ -36,6 +36,8 @@ const AppIconIndicators = Me.imports.appIconIndicators;
 
 let tracker = Shell.WindowTracker.get_default();
 
+let RecentManager = new Gtk.RecentManager();
+
```
Comment (micheleg) [inline appIcons.js]: should be `recentManager` for consistency I believe.

**Comment by micheleg** [comment]
> Faster: if Nautilus is already launched, clicking on the sidebar is faster, but otherwise it's not

I will try yout patches in the next few days. I'm curious to see how many times I will end up using those shortcuts :)

**Comment by Maestroschan** [comment]
but it's not as fast as "Places status indicator" of course

**Comment by maoschanz** [comment]
good news is that i fixed the issues, bad news is that i needed a working version so i updated the extension before fixing the issues, and it was a terrible idea since now there is a ton of conflicts (i only committed appIcons.js to avoid useless ones)

**Pull request closed without merging**